Core/App.Web/StartupBase.cs

```diff
@@ -44,6 +46,11 @@ public virtual void ConfigureServices(IServiceCollection services)
             OnSetupApplication(applicationSetup);
 
             services
+                .AddDefaultCorrelationId(config =>
```

Comment:
Could you please provide me a sample how you use the correlationId on your project? Do you add it in the header? Thanks.

Reply:
Here you can find some documentation on how it can be used: https://github.com/stevejgordon/CorrelationId/wiki

But the idea is, the front-end client generates a CorrelationId (GUID) and adds it to the **X-Correlation-Id** header. than this correlationid will be used in:
* Logging
* Audit Log
* other locations, depending on business case.

The idea is that everything is traceable to an original request. So someone does something from a front-end client, it generate a HTTP request with that correlation id, the actions that specific request triggered can be traced in the logging, audit log and maybe even other database actions (when stored).

If a request doesn't contain a correlation id, where does that request come from, since the front end should generate one. this makes these kind of requests suspicious (a good thing).

If some strange action happened in the database or something in the auditlog, you can trace it back to the original network request or logging rules, since they all contain that same correlation id.

@Johanh-od @jstorteboom

Reply:
I appreciate your answer. Many thanks.